daemon/container_test.go

```diff
@@ -0,0 +1,46 @@
+package daemon
+
+import (
+	"runtime"
+	"testing"
+
+	"github.com/docker/docker/api/types"
+	containertypes "github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/daemon/config"
+	"github.com/docker/go-connections/nat"
+	"github.com/stretchr/testify/require"
+)
+
+// This tests that a warning is returned when setting network mode to host and specifying published ports.
```

Comment:
nit: could use Godoc conventions: `s/This tests/TestContainerWarningHostAndPublishPorts tests/`